src/callbacks_step/analysis_dg2d.jl

```diff
@@ -279,30 +279,17 @@ end
 
 function analyze(::Val{:l2_divb}, du, u, t,
                  mesh::TreeMesh{2},
-                 equations::IdealGlmMhdEquations2D, dg::DGSEM, cache)
+                 equations, dg::DGSEM, cache)
     integrate_via_indices(u, mesh, equations, dg, cache, cache,
                           dg.basis.derivative_matrix) do u, i, j, element, equations,
                                                          dg, cache, derivative_matrix
         divb = zero(eltype(u))
         for k in eachnode(dg)
-            divb += (derivative_matrix[i, k] * u[6, k, j, element] +
-                     derivative_matrix[j, k] * u[7, i, k, element])
-        end
-        divb *= cache.elements.inverse_jacobian[element]
-        divb^2
-    end |> sqrt
-end
+            B1_kj, _, _ = magnetic_field(u[:, k, j, element], equations)
+            _, B2_ik, _ = magnetic_field(u[:, i, k, element], equations)
```

Comment:
This allocates a lot. Can we please use `get_node_vars` instead?

Reply:
I implemented a version of the `TreeMesh{3}` routines that uses `get_node_vars` [here](https://github.com/trixi-framework/Trixi.jl/commit/f14f0a3a30311342eaa05348a8635bd8f3ac8d95) (new branch). 

If I run the same benchmark as [above](https://github.com/trixi-framework/Trixi.jl/pull/2176#issuecomment-2494276279), I get the exact same number of allocations as before:

```julia
julia> @benchmark Trixi.analyze(Val{:l2_divb}(), $du, $u, 0.0, $mesh, $equations, $solver, $semi.cache)
BenchmarkTools.Trial: 4430 samples with 1 evaluation.
 Range (min … max):  930.719 μs …   3.922 ms  ┊ GC (min … max):  0.00% … 68.17%
 Time  (median):     979.486 μs               ┊ GC (median):     0.00%
 Time  (mean ± σ):     1.126 ms ± 599.382 μs  ┊ GC (mean ± σ):  12.17% ± 15.93%

  ██▄▃▁                                                     ▁▁▂ ▁
  █████▃▁▁▃▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▅███ █
  931 μs        Histogram: log(frequency) by time       3.73 ms <

 Memory estimate: 2.75 MiB, allocs estimate: 40976.

julia> @benchmark Trixi.analyze(Val{:linf_divb}(), $du, $u, 0.0, $mesh, $equations, $solver, $semi.cache)
BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):  51.021 μs … 101.782 μs  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     51.941 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   52.111 μs ±   2.429 μs  ┊ GC (mean ± σ):  0.00% ± 0.00%

  ▄ ▂▂ █▃▅█▁                                                   ▂
  █▄██▅██████▇▅█▅▇▇▆▅▅▇▅▅▄▆▄▆▆▅▇█▇▅▆▅▅▆▅▅▅▅▃▄▅▄▅▄▅▆▅▆▆▆▅▃▅▅▁▃▄ █
  51 μs         Histogram: log(frequency) by time        58 μs <

 Memory estimate: 0 bytes, allocs estimate: 0.
```